### PR TITLE
feat(networks): early-access "request a network" flow

### PR DESCRIPTION
**Pull request by serefyarar**
## Summary
Adds a minimal, additive "request a network" flow so any user can ask to create a network, staff review it, and it becomes a real network on approval.

- **Data change is one nullable column.** A request is a `networks` row with `request_status` (`pending` | `needs_changes`) and **no membership**, so it stays inert and hidden from discovery until staff approve it — approval clears `request_status` and adds the requester as owner. Form answers + staff review note live in the existing `metadata` jsonb (no new tables).
- **Backend:** `network-request` controller/service (`POST/GET /network-requests`, `GET /network-requests/pending`, `PATCH`/`DELETE /:id`, `POST /:id/review`), `lib/staff.ts` staff detection (`@index.network` or `STAFF_EMAILS`), approved + needs-changes email templates and a staff heads-up email, plus a discovery-query guard so request rows can never leak.
- **Web:** `RequestNetworkModal` (form + submitted state, doubles as edit/resubmit), request cards (In review / Needs changes with Update/Dismiss) and a staff review queue on the networks page, wired via a new `networkRequests` service.

## Deliberate simplifications (intentionally minimal — see request)
- No dedicated admin page; staff review is inline on the networks page and the "needs changes" note is collected via a prompt.
- No separate "Approved → Open network" card; once approved the network just appears in *My Networks* and the requester gets an approval email.

## Notes
- Staff notifications go to `STAFF_EMAILS` (comma-separated) or default to `hello@index.network`; that list also grants review access beyond `@index.network`.
- Requires running `db:migrate` to apply `0117` before deploy.

## Test plan
- [ ] Non-staff: "Create a network" opens the request form; submit shows the "Your request is in" state and an **In review** card.
- [ ] Staff (`@index.network`): see the request in the review queue; **Approve** creates the network (appears in My Networks) and emails the requester.
- [ ] Staff: **Needs changes** with a note shows a **Needs changes** card to the requester; **Update request** resubmits.
- [ ] Dismiss removes the request.
- [ ] Request rows never appear in Discover.
- [ ] `bunx tsc --noEmit` (api) clean; changed files lint clean.

### CODE REVIEW

File: apps/web/src/app/networks/page.tsx [hunk 1]
```diff
@@ -102,15 +162,20 @@ export default function NetworksPage() {
...
-              {user?.email?.endsWith('@index.network') && (
-                <button
-                  onClick={() => setCreateNetworkModalOpen(true)}
-                  className="flex items-center gap-1.5 px-3 py-1.5 text-sm text-gray-700 hover:bg-gray-100 border border-gray-200 rounded-sm transition-colors"
-                >
-                  <Plus className="w-4 h-4" />
-                  Create
-                </button>
-              )}
+              <button
```
Comment (yanekyuk) [inline apps/web/src/app/networks/page.tsx:155]: This UI sends normal users to the request form, but it does not enforce review. A normal API or CLI client can still call POST /networks and receive a live, owned network immediately. Please either enforce the early-access rule on the server (including the experiment branch) or clarify that direct API creation is intentionally exempt.

File: services/api/src/services/network-request.service.ts [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+      throw new Error('Network request not found');
+    }
+    const req = readRequest(row);
+    if (!req?.requestedByUserId) {
+      throw new Error('Network request has no requester');
+    }
+
+    if (decision === 'approve') {
+      const nextMeta = { request: { ...req, reviewedAt: new Date().toISOString(), reviewNote: undefined } };
+      const [updated] = await db
```
Comment (yanekyuk) [inline services/api/src/services/network-request.service.ts]: The request is read, its status is updated, and the owner membership is added in separate operations. Two staff reviews—or a requester update/dismiss racing a review—can overwrite each other and leave contradictory state. Please make these state transitions a single serialized transaction (or guarded compare-and-set), include the membership write, and send email only after the change commits.

**Review by yanekyuk — COMMENTED** [review]
I found two blockers in the early-access flow and left them inline. Please also add focused tests for the authorization boundary and concurrent request transitions. The staff-review UI should use a server-provided capability so STAFF_EMAILS users and mixed-case staff emails can review requests.

File: apps/web/src/app/networks/page.tsx (hunk 1 above)
Comment (serefyarar) [reply apps/web/src/app/networks/page.tsx:155]: Enforced server-side in `f3057ab6`. `POST /networks` now requires staff (`isStaff`) for **both** the community and experiment branches — non-staff API/CLI clients get a 403 pointing to `POST /network-requests`. The web UI's create-vs-request choice is now driven by a server-provided `canReview` capability returned from `GET /network-requests` (covers `STAFF_EMAILS` and mixed-case `@index.network`), not a client-side email check. Added a controller-boundary test asserting non-staff creation is blocked before any DB write.

File: services/api/src/services/network-request.service.ts (hunk 2 above)
Comment (serefyarar) [reply services/api/src/services/network-request.service.ts]: Fixed in `f3057ab6`. `reviewRequest`, `updateRequest`, and `dismissRequest` now run in a single `db.transaction` with a `FOR UPDATE` row lock, so two reviews — or a requester update/dismiss racing a review — serialize instead of overwriting each other. The owner-membership insert happens **inside** the review transaction (guarded by `onConflictDoNothing`), and the membership event + requester email fire only **after** the transaction commits. Added a source-contract test pinning the lock/inside-txn/after-commit guarantees.
